codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/StaticTable.java

```diff
@@ -31,84 +31,90 @@
+    /* 13 */ newHeaderField(":status", "404"),
+    /* 14 */ newHeaderField(":status", "500"),
+    /* 15 */ newEmptyHeaderField("accept-charset"),
+    /* 16 */ newHeaderField("accept-encoding", "gzip, deflate"),
+    /* 17 */ newEmptyHeaderField("accept-language"),
+    /* 18 */ newEmptyHeaderField("accept-ranges"),
+    /* 19 */ newEmptyHeaderField("accept"),
+    /* 20 */ newEmptyHeaderField("access-control-allow-origin"),
+    /* 21 */ newEmptyHeaderField("age"),
+    /* 22 */ newEmptyHeaderField("allow"),
+    /* 23 */ newEmptyHeaderField("authorization"),
+    /* 24 */ newEmptyHeaderField("cache-control"),
+    /* 25 */ newEmptyHeaderField("content-disposition"),
+    /* 26 */ newEmptyHeaderField("content-encoding"),
+    /* 27 */ newEmptyHeaderField("content-language"),
+    /* 28 */ newEmptyHeaderField("content-length"),
+    /* 29 */ newEmptyHeaderField("content-location"),
+    /* 30 */ newEmptyHeaderField("content-range"),
+    /* 31 */ newEmptyHeaderField("content-type"),
+    /* 32 */ newEmptyHeaderField("cookie"),
+    /* 33 */ newEmptyHeaderField("date"),
+    /* 34 */ newEmptyHeaderField("etag"),
+    /* 35 */ newEmptyHeaderField("expect"),
+    /* 36 */ newEmptyHeaderField("expires"),
+    /* 37 */ newEmptyHeaderField("from"),
+    /* 38 */ newEmptyHeaderField("host"),
+    /* 39 */ newEmptyHeaderField("if-match"),
+    /* 40 */ newEmptyHeaderField("if-modified-since"),
+    /* 41 */ newEmptyHeaderField("if-none-match"),
+    /* 42 */ newEmptyHeaderField("if-range"),
+    /* 43 */ newEmptyHeaderField("if-unmodified-since"),
+    /* 44 */ newEmptyHeaderField("last-modified"),
+    /* 45 */ newEmptyHeaderField("link"),
+    /* 46 */ newEmptyHeaderField("location"),
+    /* 47 */ newEmptyHeaderField("max-forwards"),
+    /* 48 */ newEmptyHeaderField("proxy-authenticate"),
+    /* 49 */ newEmptyHeaderField("proxy-authorization"),
+    /* 50 */ newEmptyHeaderField("range"),
+    /* 51 */ newEmptyHeaderField("referer"),
+    /* 52 */ newEmptyHeaderField("refresh"),
+    /* 53 */ newEmptyHeaderField("retry-after"),
+    /* 54 */ newEmptyHeaderField("server"),
+    /* 55 */ newEmptyHeaderField("set-cookie"),
+    /* 56 */ newEmptyHeaderField("strict-transport-security"),
+    /* 57 */ newEmptyHeaderField("transfer-encoding"),
+    /* 58 */ newEmptyHeaderField("user-agent"),
+    /* 59 */ newEmptyHeaderField("vary"),
+    /* 60 */ newEmptyHeaderField("via"),
+    /* 61 */ newEmptyHeaderField("www-authenticate")
     );
 
-    private static final Map<String, Integer> STATIC_INDEX_BY_NAME = createMap();
+    private static HeaderField newEmptyHeaderField(CharSequence name) {
+        return newHeaderField(name, AsciiString.EMPTY_STRING);
+    }
+
+    private static HeaderField newHeaderField(CharSequence name, CharSequence value) {
+        return new HeaderField(AsciiString.of(name), AsciiString.of(value));
+    }
+
+    private static final Map<CharSequence, Integer> STATIC_INDEX_BY_NAME = createMap();
```

Comment:
I think we need to use a map structure that allows us to provide our own hashCode/equals. Otherwise we may not get the expected behavior between different types of `CharSequence` objects. To accommodate for this in other areas in codec-http2 I created [CharSequenceMap](https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/CharSequenceMap.java).